### PR TITLE
Migrate Airdrop modules from KGen web to Airdrop 

### PR DESCRIPTION
**Pull request by pankaj-gkm**
- Added api-hooks for 
  -  register Api
  - resend api
  - verify api

- Added atoms for
  - loginFlow
  - modal state

- Added hooks too manage atoms  
- Added common lib files
- Added path aliases for tsconfig

### CODE REVIEW

File: modules/Authentication/hooks/login/useIndianLoginFlow.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import {modalAtom, loginFlowAtom} from '@auth-store';
+import {Flows, LOGIN_FLOW} from '@auth-lib';
+// import { INDIAN_LOGIN_FLOW } from "@auth-kgen/indianLoginFlow";
+import {useUpdateModalAtom} from '@auth-hooks';
+
+export default function useIndianLoginFlow() {
+  const {initiateFlow} = useUpdateModalAtom();
+  const {step} = useAtomValue(modalAtom);
+  // const Component = INDIAN_LOGIN_FLOW[step];
+  const Component = null;
```
Comment (shelly19gkmit) [inline modules/Authentication/hooks/login/useIndianLoginFlow.tsx:11]: why null?

File: modules/Authentication/lib/ApiCalls.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  params: AxiosRequestConfig = {},
+) => {
+  return new Promise<ApiCallResponse<T>>((resolve, reject) => {
+    instance
+      .delete<T>(url, params)
+      .then(response => resolve(successResponse(response)))
+      .catch(error => reject(failedResponse(error)));
+  });
+};
+
```
Comment (shelly19gkmit) [inline modules/Authentication/lib/ApiCalls.ts:70]: const sendRequest = <T>(
  method: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params: AxiosRequestConfig = {}
): Promise<ApiCallResponse<T>> => {
  return new Promise<ApiCallResponse<T>>((resolve, reject) => {
    instance[method]<T>(url, params)
      .then(response => resolve(successResponse(response)))
      .catch(error => reject(failedResponse(error)));
  });
};

const get = <T extends unknown>(
  url: string,
  params: AxiosRequestConfig = {}
) => sendRequest<T>('get', url, params);

const post = <T extends unknown>(
  url: string,
  params: object = {}
) => sendRequest<T>('post', url, params);

const putWithAuth = <T extends unknown>(
  url: string,
  params: AxiosRequestConfig = {}
) => sendRequest<T>('put', url, params);

const deleteWithAuth = <T extends unknown>(
  url: string,
  params: AxiosRequestConfig = {}
) => sendRequest<T>('delete', url, params);

File: package-lock.json
Comment (shelly19gkmit) [inline package-lock.json:1]: Remove package-lock.json

File: modules/Authentication/hooks/login/loginFlow/useUpdateLoginFlowAtom.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    (value: LoginFlowType[T]) =>
+      updateLoginFlowAtom({[key]: value});
+
+  const setIsLoggedIn = loginFlowSetter('isLoggedIn');
+  const setCountryCode = loginFlowSetter('countryCode');
+  const setError = loginFlowSetter('error');
+
+  // utils
+  const isIndia = isIndianUser(loginAtom.countryCode);
+  const isBrazil = isBrazilUser(loginAtom.countryCode);
```
Comment (shelly19gkmit) [inline modules/Authentication/hooks/login/loginFlow/useUpdateLoginFlowAtom.ts:28]: this won't work, fix this using locationData.countryCode -

const isIndia = isIndianUser(locationData.countryCode);
  const isBrazil = isBrazilUser(locationData.countryCode);

File: modules/Authentication/lib/ApiCalls.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  method: AxiosMethods,
+  url: string,
+  config?: AxiosRequestConfig,
+  data?: any,
+) => {
+  switch (method) {
+    case 'get':
+    case 'delete':
+      return instance[method]<T>(url, config);
+
```
Comment (shelly19gkmit) [inline modules/Authentication/lib/ApiCalls.ts:40]: remove extra blank line

File: modules/Authentication/lib/ApiCalls.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      .catch(error => reject(failedResponse(error)));
+  });
+}
+
+function methodBuilder(
+  method: 'get' | 'delete',
+): <R>(url: string, config?: AxiosRequestConfig) => Promise<ApiCallResponse<R>>;
+
+function methodBuilder(
+  method: 'post' | 'put',
```
Comment (shelly19gkmit) [inline modules/Authentication/lib/ApiCalls.ts:67]: create constants for post, put, delete, get

File: modules/Authentication/lib/index.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export {default as apiCalls} from './ApiCalls';
+export {default as instance} from "./instance"
+export * from './constants';
+export * from './types';
+export * from './ApiCalls';
+export * from './steps';
+export * from './regex';
+// export * from "./utils";
```
Comment (shelly19gkmit) [inline modules/Authentication/lib/index.ts:8]: is this comment needed?

File: modules/Authentication/hooks/login/loginFlow/useUpdateLoginFlowAtom.ts (hunk 3 above)
Comment (pankaj-gkm) [reply modules/Authentication/hooks/login/loginFlow/useUpdateLoginFlowAtom.ts:28]: location provider is not yet implemented in airdrop

File: modules/Authentication/lib/ApiCalls.ts (hunk 4 above)
Comment (pankaj-gkm) [reply modules/Authentication/lib/ApiCalls.ts:40]: have kept a single blank line between different case

File: modules/Authentication/hooks/login/loginFlow/useUpdateLoginFlowAtom.ts (hunk 3 above)
Comment (shelly19gkmit) [reply modules/Authentication/hooks/login/loginFlow/useUpdateLoginFlowAtom.ts:28]: Add locationData in store/login.ts for now, to make everything consistent

File: modules/Authentication/hooks/login/useOtpLessFlow.tsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
+import {
+  AB_CONSTS,
+  OTP_ENABLED_COUNTRIES,
+  OTPLESS_AUTH_SCRIPT,
+} from '../../lib/constants';
```
Comment (shelly19gkmit) [inline modules/Authentication/hooks/login/useOtpLessFlow.tsx:5]: use Alias

File: modules/Authentication/utils/index.ts [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  //   case Flows.Login:
+  //     return INDIAN_LOGIN_FLOW;
+  //   case Flows.DeleteClan:
+  //     return INDIAN_DELETE_CLAN_FLOW;
+  //   default:
+  //     return {};
+  // }
+};
+
+export const getStyles = (flowType?: Flows) => {
```
Comment (shelly19gkmit) [inline modules/Authentication/utils/index.ts:54]: Add comment

**Pull request merged**